copy.go

```diff
@@ -54,6 +56,40 @@ type CopyOptions struct {
 	MapRoot func(ctx context.Context, src content.Storage, root ocispec.Descriptor) (ocispec.Descriptor, error)
 }
 
+// selectPlatform implements platform filter and returns the descriptor of
+// the first matched manifest from the manifest list / image index.
+func (o *CopyOptions) selectPlatform(ctx context.Context, src content.Storage, root ocispec.Descriptor, p *ocispec.Platform) (ocispec.Descriptor, error) {
+	if root.MediaType != docker.MediaTypeManifestList && root.MediaType != ocispec.MediaTypeImageIndex {
+		return ocispec.Descriptor{}, errdef.ErrUnsupported
+	}
+
+	manifests, err := content.Successors(ctx, src, root)
+	if err != nil {
+		return ocispec.Descriptor{}, err
+	}
+
+	// platform filter
+	for _, m := range manifests {
+		if platform.MatchPlatform(m.Platform, p) {
+			return m, nil
+		}
+	}
+	return ocispec.Descriptor{}, errdef.ErrNotFound
+}
+
+// AddPlatformFilter adds the selectPlatform func into the MapRoot func.
```

Comment:
This is user-facing documentation; hence we should not mention any internal function names. We should explain what's the effect of this function.